### PR TITLE
Remap Windows targets triples to their LLVM counterparts

### PR DESCRIPTION
**Pull request by mati865**
Fixes https://github.com/rust-lang/cc-rs/issues/1167

### CODE REVIEW

File: src/target_info.rs [hunk 1]
```diff
@@ -5,10 +5,21 @@ pub const RISCV_ARCH_MAPPING: &[(&str, &str)] = &[
     ("riscv32gc", "riscv32"),
     ("riscv32i", "riscv32"),
     ("riscv32im", "riscv32"),
-    ("riscv32ima", "riscv32"),
```
Comment (NobodyXu) [inline src/target_info.rs:8]: Is this deletion accidental?

File: src/lib.rs [hunk 2]
```diff
@@ -3584,11 +3595,11 @@ impl Build {
...
-            // If no toolchain was found, provide the first toolchain that was passed in.
-            // This toolchain has been shown not to exist, however it will appear in the
-            // error that is shown to the user which should make it easier to search for
-            // where it should be obtained.
-            .or_else(|| prefixes.first().copied())
+        // If no toolchain was found, provide the first toolchain that was passed in.
+        // This toolchain has been shown not to exist, however it will appear in the
+        // error that is shown to the user which should make it easier to search for
+        // where it should be obtained.
+        .or_else(|| prefixes.first().copied())
```
Comment (NobodyXu) [inline src/lib.rs]: I think you need to revert this change as it breaks cargo-fmt pipeline

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: src/target_info.rs [hunk 3]
```diff
@@ -5,10 +5,21 @@ pub const RISCV_ARCH_MAPPING: &[(&str, &str)] = &[
...
     ("riscv32imac", "riscv32"),
     ("riscv32imafc", "riscv32"),
     ("riscv32imc", "riscv32"),
     ("riscv64gc", "riscv64"),
     ("riscv64imac", "riscv64"),
 ];
+pub const WINDOWS_TRIPLE_MAPPING: &[(&str, &str)] = &[
```
Comment (mati865) [inline src/target_info.rs:14]: No idea if I should skip `uwp` and `win7` here.

File: src/target_info.rs [hunk 4]
```diff
@@ -5,10 +5,21 @@ pub const RISCV_ARCH_MAPPING: &[(&str, &str)] = &[
...
+    ("aarch64-pc-windows-gnullvm", "aarch64-pc-windows-gnu"),
+    ("aarch64-uwp-windows-msvc", "aarch64-pc-windows-msvc"),
+    ("i686-pc-windows-gnullvm", "i686-pc-windows-gnu"),
+    ("i686-uwp-windows-gnu", "i686-pc-windows-gnu"),
+    ("i686-uwp-windows-msvc", "i686-pc-windows-msvc"),
+    ("i686-win7-windows-msvc", "i686-pc-windows-msvc"),
```
Comment (mati865) [inline src/target_info.rs:20]: @roblabla i686 and x86_64 have different vendor in their `llvm_triple`, is it intended?
https://github.com/rust-lang/rust/pull/118150/files#diff-bdd6b94a9eff21c9a3c3d1bb987409f8c4f2fc86c3ba7fccaa24759dd46975e9R11

File: src/target_info.rs (hunk 1 above)
Comment (mati865) [reply src/target_info.rs]: At first I thought this array was out of date but now realised I've been running this tool with stable toolchain...

File: src/lib.rs (hunk 2 above)
Comment (mati865) [reply src/lib.rs]: Yeah, JetBrains formatter is kinda broken sometimes.

File: src/target_info.rs (hunk 4 above)
Comment (NobodyXu) [reply src/target_info.rs:21]: Not sure cc @ChrisDenton can you take a look at this please?

Comment (ChrisDenton) [reply src/target_info.rs:21]: That looks like a rustc bug to me. LLVM does not abuse the vendor field like rustc does so it should always be `pc`.

Comment (roblabla) [reply src/target_info.rs:21]: Ah yeah that's on me, it should definitely say pc in there.

File: src/target_info.rs (hunk 3 above)
Comment (NobodyXu) [reply src/target_info.rs:15]: I think it's fine

**Review by NobodyXu — APPROVED** [review]
Thanks, LGTM!

File: src/target_info.rs (hunk 4 above)
Comment (kleisauke) [reply src/target_info.rs:21]: I just opened PR https://github.com/rust-lang/rust/pull/133239 for this.
